include/lsst/geom/Box.h

```
@@ -104,6 +106,22 @@ class Box2I {
     Box2I(Box2I&&) = default;
     ~Box2I() = default;
 
+    /**
+     * Create a box centered as closely as possible on a particular point.
+     *
+     * @param center The desired center of the box.
+     * @param size The desired width and height (in that order) of the box.
+     * @param invert If true (default), invert any negative dimensions
+     *               instead of creating an empty box.
+     *
+     * @returns if `size` is positive, a box with size `size`; otherwise,
+     *          an empty box or one with modified dimensions following the
+     *          conventions of Box2I(Point2I const&, Extent2I const&, bool).
+     *          If the returned box is not empty, its center shall be within
+     *          half a pixel of `center` in either dimension.
+     */
+    static Box2I makeCenteredBox(Point2D const& center, Extent const& size, bool invert = true);
```

There's actually an [adopted RFC](https://jira.lsstcorp.org/browse/RFC-324) for changing the default for `invert` in the constructors to `false`.  That work hasn't been done primarily because it's not backwards-compatible, and hence requires some care and effort to track down breakage.  Since this is a new API it might be better to use `invert=false` from the start or even drop that argument from this signature entirely, unless you have a use case for it.

---

I think if we're going to mix factory methods and constructors in APIs, we should require that they behave consistently. Anything else will lead to user error.

I could set `invert=false`, but by the same token I'd be worried if the RFC didn't get implemented quickly and we were stuck with inconsistent behavior for an extended period of time.

---

What about just removing it?  While I am concerned about blithely changing the default without taking care to avoid breakage, I think it's also true that `invert` is essentially never used, and if we don't have it here there'd be no expectation of consistency.

---

Actually, I consider removing it the worst option, barring an RFC to remove `invert` from the constructors as well.

The problem with defaults is that they tend to hide parameters from users. In this case, a developer who prefers to learn APIs from example rather than by consulting the documentation could well know that `Box(myPoint, Extent(-3, -4))` returns a 3×4 box without knowing that `Box(myPoint, Extent(-3, -4), false)` is even a valid call. To such a developer, the absence of an `invert` parameter in `makeCenteredBox` would by no means imply that it interprets dimensions differently.

If we make `invert=false` by default, that wouldn't fix the problem of surprising default behavior, but at least the user would have an easy workaround once they realized what was going on.

---

I do think not having `invert` is the right long-term signature for both the constructors and this function, and I have a slight preference for doing whatever makes it easiest to migrate to that state.  But I won't object if you'd rather emphasize consistency in the current interface either a little (by retaining `invert` and making it default to `False`) or a lot (by retaining it with a default of `true`).  The future and the present are both worthwhile things to emphasize.